src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/gc/g1/G1CollectedHeap.java

```diff
@@ -90,42 +90,42 @@ public long n_regions() {
         return hrm().length();
     }
 
-    public HeapRegionManager hrm() {
+    public G1HeapRegionManager hrm() {
         Address hrmAddr = addr.addOffsetTo(hrmFieldOffset);
-        return VMObjectFactory.newObject(HeapRegionManager.class, hrmAddr);
+        return VMObjectFactory.newObject(G1HeapRegionManager.class, hrmAddr);
     }
 
     public G1MonitoringSupport monitoringSupport() {
         Address monitoringSupportAddr = monitoringSupportField.getValue(addr);
         return VMObjectFactory.newObject(G1MonitoringSupport.class, monitoringSupportAddr);
     }
 
-    public HeapRegionSetBase oldSet() {
+    public G1HeapRegionSetBase oldSet() {
         Address oldSetAddr = addr.addOffsetTo(oldSetFieldOffset);
-        return VMObjectFactory.newObject(HeapRegionSetBase.class, oldSetAddr);
+        return VMObjectFactory.newObject(G1HeapRegionSetBase.class, oldSetAddr);
     }
 
-    public HeapRegionSetBase humongousSet() {
+    public G1HeapRegionSetBase humongousSet() {
         Address humongousSetAddr = addr.addOffsetTo(humongousSetFieldOffset);
-        return VMObjectFactory.newObject(HeapRegionSetBase.class, humongousSetAddr);
+        return VMObjectFactory.newObject(G1HeapRegionSetBase.class, humongousSetAddr);
     }
 
-    private Iterator<HeapRegion> heapRegionIterator() {
+    private Iterator<G1HeapRegion> heapRegionIterator() {
         return hrm().heapRegionIterator();
     }
 
-    public void heapRegionIterate(HeapRegionClosure hrcl) {
-        Iterator<HeapRegion> iter = heapRegionIterator();
+    public void heapRegionIterate(G1HeapRegionClosure hrcl) {
+        Iterator<G1HeapRegion> iter = heapRegionIterator();
         while (iter.hasNext()) {
-            HeapRegion hr = iter.next();
+            G1HeapRegion hr = iter.next();
             hrcl.doHeapRegion(hr);
         }
     }
 
-    public HeapRegion heapRegionForAddress(Address addr) {
-        Iterator<HeapRegion> iter = heapRegionIterator();
+    public G1HeapRegion heapRegionForAddress(Address addr) {
+        Iterator<G1HeapRegion> iter = heapRegionIterator();
         while (iter.hasNext()) {
-            HeapRegion hr = iter.next();
+            G1HeapRegion hr = iter.next();
             if (hr.isInRegion(addr)) {
                 return hr;
             }
```

Comment:
Since these three methods are G1 specific, I'd prefer not to add the 'g1' prefix.